### ISSUE TITLE
VPCFlow configuration is not correct

### ISSUE THREAD

**Issue by mgmacias95** [sonofagl1tch/AWSDetonationLab#74]
Hello,

The VPCFlow Bucket where logs are stored is being configured as `custom` type but it must be `vpflow`:
https://github.com/sonofagl1tch/AWSDetonationLab/blob/f179d8b44f60d0eb1424d3f72ef4b3254f55ba20/additionalInstallationScripts/installWazuh.sh#L174-L179

This makes VPC Flow logs not being parsed/processed by Wazuh.

Best regards,
Marta

**Comment by sonofagl1tch** [comment]
is the vpflow variable what wazuh uses or was that a typo? the type should be vpcflow right?

**Comment by sonofagl1tch** [comment]
confirmed with [wazuh documentation](https://documentation.wazuh.com/3.x/amazon/services/vpc.html)  that the bucket type is vpcflow and not vpflow.

**Comment by sonofagl1tch** [comment]
Not sure why I am getting a SIGINT error for the vpcflow bucket. Any ideas for common things to troubleshoot?


Code | Description
-- | --
1 | Unknown error
2 | Error parsing configuration (bucket name, keys, etc)
3 | Invalid credentials to access S3 bucket
4 | boto3 module missing
5 | Unexpected error accessing SQLite DB
6 | Unable to create SQLite DB
7 | Unexpected error querying/working with objects in S3
8 | Failed to decompress file
9 | Failed to parse file
10 | Failed to execute DB cleanup
11 | Unable to connect to Wazuh
12 | SIGINT
13 | Error sending message to Wazuh



> 2019/03/26 01:51:04 wazuh-modulesd:aws-s3: INFO: Executing Bucket Analysis: fixvpcflowconfig-s3bucketvpcflow-en1aeaoakcwb
2019/03/26 01:51:04 wazuh-modulesd:aws-s3: WARNING: Bucket:  -  Returned exit code 12
2019/03/26 01:51:04 wazuh-modulesd:aws-s3: WARNING: Bucket:  -  CommonPrefixes
2019/03/26 01:51:04 wazuh-modulesd:aws-s3: INFO: Fetching logs finished.

**Comment by mgmacias95** [comment]
Hello @sonofagl1tch,

After doing some research I found out this is how a VPCFlow bucket is structured:
```
2019-03-27 10:32:41       1568 AWSLogs/123456789/vpcflowlogs/us-east-1/2019/03/27/123456789_vpcflowlogs_us-east-1_fl-0c9509c13d9508928_20190327T1030Z_7801f117.log.gz
```

Our current configuration is expecting logs to be placed under `flowlogs/` directory:
https://github.com/sonofagl1tch/AWSDetonationLab/blob/2aafd119383dba81e565f14dbbc7ff5783efe70f/additionalInstallationScripts/installWazuh.sh#L175-L180
Removing that option from the configuration should fix the issue 😄 .

Best regards,
Marta

**Comment by sonofagl1tch** [comment]
> Hello @sonofagl1tch,
> 
> After doing some research I found out this is how a VPCFlow bucket is structured:
> 
> ```
> 2019-03-27 10:32:41       1568 AWSLogs/123456789/vpcflowlogs/us-east-1/2019/03/27/123456789_vpcflowlogs_us-east-1_fl-0c9509c13d9508928_20190327T1030Z_7801f117.log.gz
> ```
> Our current configuration is expecting logs to be placed under `flowlogs/` directory:
> [AWSDetonationLab/additionalInstallationScripts/installWazuh.sh](https://github.com/sonofagl1tch/AWSDetonationLab/blob/2aafd119383dba81e565f14dbbc7ff5783efe70f/additionalInstallationScripts/installWazuh.sh#L175-L180)
> 
> Lines 175 to 180 in [2aafd11](/sonofagl1tch/AWSDetonationLab/commit/2aafd119383dba81e565f14dbbc7ff5783efe70f)
> 
>      <bucket type="custom"> 
>        <name>vpcflowlogging</name> 
>        <path>flowlogs</path> 
>        <access_key>insert_access_key</access_key> 
>        <secret_key>insert_secret_key</secret_key> 
>      </bucket> 
> 
> Removing that option from the configuration should fix the issue 😄 .
> Best regards,
> Marta

so vpcflow doesnt need `<path>flowlogs</path> ` in the config file? should it be changed to something new or just removed entirely? also, was this a change to wazuh or something else?

**Comment by mgmacias95** [comment]
Removed entirely. This change was fixed here in [#21](https://github.com/sonofagl1tch/AWSDetonationLab/pull/21/files#diff-8b35691ac62cf9abb2671fd61cd8df1aL87) but it got lost in the last refactor we did of the installWazuh script.
